__tests__/components/Calculator/BestSubject.test.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+
+import BestSubject from '../../../src/components/Calculator/BestSubject'
+
+const setup = (setupProps = {}) => {
+  const defaultProps = {
+    bestSubject: {
+      name: '',
+      maxScore: {
+        value: 0,
+        formula: ''
+      }
+    },
+    subjectsType: 'primary'
+  }
+  const props = { ...defaultProps, ...setupProps }
+  const wrapper = shallow(<BestSubject {...props} />)
+
+  return {
+    props,
+    wrapper
+  }
+}
+
+describe('BestSubject component', () => {
+  test('should render', () => {
+    const { wrapper } = setup()
+
+    expect(wrapper).toMatchSnapshot()
+  })
+
+  test('should render "bez nazwy" when no subject name provided', () => {
```

Comment:
The following test cases are very precise ;) Most of that can be handled via matching snapshots.
We'll discuss this in the chat.